### PR TITLE
Custom health bars

### PR DESCRIPTION
**Pull request by Alias-Alan**
HealthBarGumpCustom with option between classic/custom added into experimental.

### CODE REVIEW

File: src/Renderer/RenderedText.cs [hunk 1]
```diff
@@ -285,5 +290,15 @@ public void Destroy()
 
             _pool.Enqueue(this);
         }
+        internal Action<object, MouseEventArgs> ToString(object mouseUp)
```
Comment (andreakarasho) [inline src/Renderer/RenderedText.cs]: what is this?

File: src/Renderer/RenderedText.cs [hunk 2]
```diff
@@ -285,5 +290,15 @@ public void Destroy()
...
+        {
+            throw new NotImplementedException();
+        }
+
+        internal Action<object, MouseEventArgs> ToString(TextBox MouseUp)
```
Comment (andreakarasho) [inline src/Renderer/RenderedText.cs]: what is this? (x2)

File: src/Game/UI/Gumps/HealthBarGumpCustom.cs [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+            protected set { }
+        }
+
+        public override int GroupMatrixHeight
+        {
+            get => Height;
+            protected set { }
+        }
+
+        public void Update()
```
Comment (andreakarasho) [inline src/Game/UI/Gumps/HealthBarGumpCustom.cs:140]: can you rename this in ForceUpdate() ?
This is a standard behaviour that im using now.

File: src/Game/UI/Gumps/HealthBarGumpCustom.cs [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+        private TextBox _textBox;
+
+        public RenderedText _renderedText;
+        public AlphaBlendControl _background;
+
+        public HealthBarGumpCustom(Entity entity) : this()
+        {
+            if (Engine.Profile.Current.CustomBarsToggled == false)
+            {
+                Engine.UI.GetGump<HealthBarGump>(entity.Serial);
```
Comment (andreakarasho) [inline src/Game/UI/Gumps/HealthBarGumpCustom.cs]: What's happening here?
You get the object and...? Missing a dispose maybe?

File: src/Game/Scenes/GameSceneInputHandler.cs [hunk 5]
```diff
@@ -706,8 +825,11 @@ internal override void OnKeyDown(SDL.SDL_KeyboardEvent e)
             if (TargetManager.IsTargeting && e.keysym.sym == SDL.SDL_Keycode.SDLK_ESCAPE && Keyboard.IsModPressed(e.keysym.mod, SDL.SDL_Keymod.KMOD_NONE))
                 TargetManager.CancelTarget();
 
-            if (!Engine.UI.IsKeyboardFocusAllowHotkeys)
-                return;
+            if (Engine.Profile.Current.ActivateChatAfterEnter)
```
Comment (roxya) [inline src/Game/Scenes/GameSceneInputHandler.cs]: This is a revert of Mel's quite amazing keyboard focus improvements?

Comment (Alias-Alan) [reply src/Game/Scenes/GameSceneInputHandler.cs]: Might be, I can revise this.

File: src/Renderer/RenderedText.cs (hunk 2 above)
Comment (Alias-Alan) [reply src/Renderer/RenderedText.cs]: was old code/ edited to new

File: src/Game/UI/Gumps/HealthBarGumpCustom.cs (hunk 4 above)
Comment (Alias-Alan) [reply src/Game/UI/Gumps/HealthBarGumpCustom.cs]: It's to double-check to make sure it's not healthbargump, the dispose is a line down.

File: src/Game/UI/Gumps/HealthBarGumpCustom.cs (hunk 3 above)
Comment (Alias-Alan) [reply src/Game/UI/Gumps/HealthBarGumpCustom.cs:140]: Sure thing, edited it to ForceUpdate()

File: src/Renderer/RenderedText.cs (hunk 1 above)
Comment (Alias-Alan) [reply src/Renderer/RenderedText.cs]: old code

File: src/Game/UI/Gumps/HealthBarGumpCustom.cs (hunk 4 above)
Comment (andreakarasho) [reply src/Game/UI/Gumps/HealthBarGumpCustom.cs]: But that code does litterally nothing, i think you can remove it

**Review by andreakarasho — COMMENTED** [review]
I'm really curious to figure out why your GIT touches other project classes. Wtf?!

**Review by Alias-Alan — COMMENTED** [review]
Commented the get gump out and moved the dispose in it's place.

File: src/ClassicUO.csproj [hunk 6]
```diff
@@ -585,4 +586,4 @@
     <Folder Include="Renderer\UI\Layouts\" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-</Project>
\ No newline at end of file
+</Project>
```
Comment (binff04) [inline src/ClassicUO.csproj:589]: Changes like this should not be committed. Please, adjust the settings of your editor to not add newlines automatically at the end of each file you hit save on. It should be done for the entire codebase at once or we will keep as it is even if it is ugly. :)

Comment (Alias-Alan) [reply src/ClassicUO.csproj:589]: Definitely agree, pulling now, running into issue with party atm.

**Pull request closed without merging**